python/pyspark/sql/udf.py

```diff
@@ -157,7 +157,17 @@ def _create_judf(self):
         spark = SparkSession.builder.getOrCreate()
         sc = spark.sparkContext
 
-        wrapped_func = _wrap_function(sc, self.func, self.returnType)
+        func = fail_on_stopiteration(self.func)
+
+        # prevent inspect to fail
+        # e.g. inspect.getargspec(sum) raises
+        # TypeError: <built-in function sum> is not a Python function
```

Comment:
nit: seems this comment should be belonging to under `except TypeError`. 
Here an appropriate comment is why we need get argspec into `_argspec`.